internal/trace/transport.go

```diff
@@ -87,3 +92,51 @@ func logHeader(header http.Header) string {
 	}
 	return "   Empty header"
 }
+
+// logResponseBody prints out the response body if it is printable and within
+// the size limit.
+func logResponseBody(resp *http.Response) string {
+	if resp.Body == nil || resp.Body == http.NoBody || resp.ContentLength == 0 {
+		return "   No response body to print"
+	}
+
+	// non-applicable body is not printed and remains untouched for subsequent processing
+	contentType := resp.Header.Get("Content-Type")
+	if !isPrintableContentType(contentType) {
+		return fmt.Sprintf("   Response body of content type \"%s\" is not printed", contentType)
+	}
+	if resp.ContentLength < 0 {
+		return "   Response body of unknown content length is not printed"
+	}
+	if resp.ContentLength > payloadSizeLimit {
+		return fmt.Sprintf("   Response body larger than %d bytes is not printed", payloadSizeLimit)
+	}
+
+	// Note: If the actual body size mismatches the content length and exceeds the limit,
+	// the body will be truncated to the limit for seucrity consideration.
+	// In this case, the response processing subsequent to logging might be broken.
+	rc := resp.Body
+	defer rc.Close()
+	lr := io.LimitReader(rc, payloadSizeLimit)
+	bodyBytes, err := io.ReadAll(lr)
+	if err != nil {
+		return fmt.Sprintf("   Error reading response body: %v", err)
+	}
+
+	// reset the body for subsequent processing
+	resp.Body = io.NopCloser(bytes.NewReader(bodyBytes))
+	return string(bodyBytes)
+}
+
+// isPrintableContentType returns true if the content of contentType is printable.
+func isPrintableContentType(contentType string) bool {
+	// JSON types
+	if strings.HasPrefix(contentType, "application/json") || strings.HasSuffix(contentType, "+json") {
```

Comment:
`application/jsonwhatever` is not JSON.
```suggestion
	if contentType == "application/json" || strings.HasSuffix(contentType, "+json") {
```